dht.go

```diff
@@ -197,32 +205,20 @@ func (dht *IpfsDHT) getValueOrPeers(ctx context.Context, p peer.ID, key string)
 		return nil, peers, nil
 	}
 
-	log.Warning("getValueOrPeers: routing.ErrNotFound")
 	return nil, nil, routing.ErrNotFound
 }
 
 // getValueSingle simply performs the get value RPC with the given parameters
-func (dht *IpfsDHT) getValueSingle(ctx context.Context, p peer.ID, key string) (*pb.Message, error) {
-	meta := logging.LoggableMap{
-		"key":  key,
-		"peer": p,
-	}
-
-	eip := log.EventBegin(ctx, "getValueSingle", meta)
-	defer eip.Done()
+func (dht *IpfsDHT) getValueSingle(ctx context.Context, p peer.ID, key string) (_ *pb.Message, _err error) {
+	ctx = dht.logSpan(ctx, "getValueSingle")
+	defer func() {
+		log.FinishWithErr(ctx, _err)
+	}()
+	log.SetTag(ctx, "peer", p)
+	log.SetTag(ctx, "key", loggableKey(key))
```

Comment:
so much work to get a log!

Reply:
There has to be a better pattern for this, this is just a disgusting amount of work and boilerplate just to get a damn log :)